client/me/purchases/manage-purchase/payment-method-selector/assignment-processor-functions.ts

```diff
@@ -177,9 +180,8 @@ export async function assignExistingCardProcessor(
 		if ( ! purchase ) {
 			throw new Error( 'Cannot assign PayPal payment method without a purchase' );
 		}
-		return wpcomAssignPaymentMethod( String( purchase.id ), storedDetailsId ).then( ( data ) => {
```

Comment:
This was an existing bug! If `wpcomAssignPaymentMethod` rejected, it would not be caught by the `catch` below without using `await`.